src/Lazy/split.ts

```diff
@@ -0,0 +1,35 @@
+import { isIterable } from "../_internal/utils";
+
+function* sync(sep: string, str: Iterable<string>) {
+  let acc = [];
+  for (const chr of str) {
+    if (chr === sep) {
+      yield acc.join("");
+      acc = [];
+    } else if (sep === "") {
+      yield chr;
+      acc = [];
+    } else {
+      acc.push(chr);
+    }
+  }
+
+  if (acc.length > 0) {
+    yield acc.join("");
+  }
+}
+
+function split(
+  sep: string,
+  iterable: Iterable<string>,
+): IterableIterator<string>;
+
+function split(sep: string, str: Iterable<string>) {
+  if (isIterable(str)) {
+    return sync(sep, str);
+  }
+
+  throw new TypeError("iterable must be type of Iterable or AsyncIterable");
```

Comment:
Let's start with capital letters.
```suggestion
  throw new TypeError("Iterable must be type of Iterable or AsyncIterable");
```

Reply:
Thank you for your comment:)

I was thinking of the variable iterable of the current context, but I think I gave the confusion to [Iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#the_iterable_protocol) 

Should I usually express the variable expression like `iterable`?